### ISSUE TITLE
empty-heading-matches should use getRole() to account for presentation conflict resolution

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3482]
The following should fail the `empty-heading` rule, it is instead considered inapplicable because of role=none. However, role=none is in practice ignored because of the use of aria-label. As a global attribute, aria-label tells browsers to ignore role=none. This can be solved by updating the rules/heading-matches.js, replacing its custom role computation with the `aria.getRole()` commons method, which already account for presentational role conflict resolution.

```html
<h1 aria-label="" role="none"></h1>
```

**Comment by dbowling** [comment]
@WilcoFiers 

`aria.getRole()` replaced the custom logic as part of PR #3582.

I introduced two new tests to cover this case there since it was changing anyway:
- [Integration](https://github.com/dequelabs/axe-core/pull/3582/files#diff-1bcc001009df0404cb3be6a5307da75b0a033c1666f29bae9bd8eabaf6a27a5fR133)
- [Unit](https://github.com/dequelabs/axe-core/pull/3582/files#diff-ee05efa4ef695f44f20c76819bf4bd12be53d428e53cfff7ec8433d1574b3722R47)

**Comment by dbowling** [comment]
@straker The PR that references this has been merged, so I think this issue can be closed.

**Comment by padmavemulapati** [comment]
Validated with the latest develop branch code base, 
seeing `empty-heading` failure on `<h1 aria-label="" role="none"></h1>` , when role=none. (using latest axe-core rule4.4.3)
With the old rule set (4.3.5), not seeing any failure on `empty-heading` with the said snippet.
![image](https://user-images.githubusercontent.com/47409579/187220478-d22aa470-7fc9-45c6-a24a-433d81db2e17.png)
